src/components/molecules/search-bar.tsx

```diff
@@ -22,13 +22,12 @@ const SearchBar: React.FC = () => {
   useHotkeys("ctrl+k", toggleSearch, {}, [])
   useHotkeys("/", toggleSearch, {}, [])
 
+  const navigation = useNavigation()
   React.useEffect(() => {
-    return globalHistory.listen(({ action }) => {
-      if (action === "PUSH") {
-        closeModal()
-      }
-    })
-  }, [])
+    if (navigation.state === "submitting") {
+      closeModal()
+    }
+  }, [navigation.state])
```

Comment:
Bug: The topbar search modal doesn't not close upon clicking on a search result.